waveform_editor/gui/io/manager.py

```diff
@@ -55,27 +58,60 @@ def create_new_file(self):
         self.main_gui.load_yaml(yaml_content)
         self.open_file = None
 
+    def _confirm_and_execute(self, action, message):
+        def proceed():
+            action()
+
+        if self.main_gui.config.has_changed:
+            self.main_gui.confirm_modal.show(message, on_confirm=proceed)
+        else:
+            proceed()
```

Comment:
The following should work, no?

```suggestion
        if self.main_gui.config.has_changed:
            self.main_gui.confirm_modal.show(message, on_confirm=action)
        else:
            action()
```